File: sdk/php/src/Resources/AutomationResource.php
<?php

declare(strict_types=1);

namespace Zaptura\Resources;

use Zaptura\Http\HttpExecutor;

/**
 * Automation resource — autoreply rules matching inbound messages.
 *
 * Backed by src/modules/automation/automation-rule.controller.ts
 * (@Controller('sessions/:sessionId/automation-rules')).
 */
class AutomationResource
{
    private HttpExecutor $http;

    public function __construct(HttpExecutor $http)
    {
        $this->http = $http;
    }

    /** @return array<int,array<string,mixed>> */
    public function list(string $sessionId): array
    {
        return $this->http->request('GET', "/api/sessions/{$this->http->encodeSegment($sessionId)}/automation-rules") ?? [];
    }

    /** @return array<string,mixed> */
    public function get(string $sessionId, string $ruleId): array
    {
        return $this->http->request('GET', "/api/sessions/{$this->http->encodeSegment($sessionId)}/automation-rules/{$this->http->encodeSegment($ruleId)}");
    }

    /**
     * Create an autoreply rule.
     *
     * @param array<string,mixed> $body Must contain 'name' and 'replyText'.
     * @return array<string,mixed>
     */
    public function create(string $sessionId, array $body): array
    {
        return $this->http->request('POST', "/api/sessions/{$this->http->encodeSegment($sessionId)}/automation-rules", [], $body);
    }

    /**
     * Update an autoreply rule.
     *
     * @param array<string,mixed> $body
     * @return array<string,mixed>
     */
    public function update(string $sessionId, string $ruleId, array $body): array
    {
        return $this->http->request('PUT', "/api/sessions/{$this->http->encodeSegment($sessionId)}/automation-rules/{$this->http->encodeSegment($ruleId)}", [], $body);
    }

    /** Delete an autoreply rule. */
    public function delete(string $sessionId, string $ruleId): void
    {
        $this->http->request('DELETE', "/api/sessions/{$this->http->encodeSegment($sessionId)}/automation-rules/{$this->http->encodeSegment($ruleId)}");
    }
}

File: sdk/java/src/main/java/com/asmsafone/zaptura/resources/AutomationResource.java
package com.asmsafone.zaptura.resources;

import static com.asmsafone.zaptura.http.Http.encodeSegment;

import com.asmsafone.zaptura.ZapturaClient;
import com.asmsafone.zaptura.http.HttpMethod;
import com.asmsafone.zaptura.model.AutomationRule;
import com.asmsafone.zaptura.model.CreateAutomationRuleRequest;
import com.asmsafone.zaptura.model.UpdateAutomationRuleRequest;
import java.util.List;

/** Automation resource — autoreply rules matching inbound messages. */
public final class AutomationResource {
    private final ZapturaClient client;

    public AutomationResource(ZapturaClient client) {
        this.client = client;
    }

    /** List all automation rules for a session. */
    public List<AutomationRule> list(String sessionId) {
        return client.requestList(
            HttpMethod.GET,
            "/api/sessions/" + encodeSegment(sessionId) + "/automation-rules",
            null,
            null,
            AutomationRule.class);
    }

    /** Get a single automation rule by ID. */
    public AutomationRule get(String sessionId, String ruleId) {
        return client.request(
            HttpMethod.GET,
            "/api/sessions/" + encodeSegment(sessionId) + "/automation-rules/" + encodeSegment(ruleId),
            null,
            null,
            AutomationRule.class);
    }

    /** Create an automation rule. */
    public AutomationRule create(String sessionId, CreateAutomationRuleRequest body) {
        return client.request(
            HttpMethod.POST,
            "/api/sessions/" + encodeSegment(sessionId) + "/automation-rules",
            null,
            body,
            AutomationRule.class);
    }

    /** Update an automation rule. */
    public AutomationRule update(String sessionId, String ruleId, UpdateAutomationRuleRequest body) {
        return client.request(
            HttpMethod.PUT,
            "/api/sessions/" + encodeSegment(sessionId) + "/automation-rules/" + encodeSegment(ruleId),
            null,
            body,
            AutomationRule.class);
    }

    /** Delete an automation rule. */
    public void delete(String sessionId, String ruleId) {
        client.request(
            HttpMethod.DELETE,
            "/api/sessions/" + encodeSegment(sessionId) + "/automation-rules/" + encodeSegment(ruleId),
            null,
            null,
            Void.class);
    }
}
